### ISSUE TITLE
False branch of in-operator type guard for non-union intersection is not never

### ISSUE THREAD

**Issue by pizzacat83** [microsoft/TypeScript#37039]
**TypeScript Version:**  Nightly

**Search Terms:** intersection guard, intersection never

**Expected behavior:** In the first function, the type of `obj` in the `else` block is `never`

**Actual behavior:** In the first function, the type of `obj` in the `else` block is `{ str: string } & { num: number }`, the same as in the `if` block.

**Related Issues:** I didn't find any.

**Code**
```ts
(obj: { str: string } & { num: number }) => {
    // Non-Union Intersection type doesn't work as expected.
    if ('str' in obj) {
        const s: string = obj.str;
        // obj: { str: string } & { num: number }
    } else {
        const _exhaustiveCheck: never = obj;
        // Expected: No error. obj: never
        // Actual: Type '{ str: string; } & { num: number; }' is not assignable to type 'never'.
    }
};

(obj: { str: string; num: number }) => {
    // Non-intersection type works as expected.
    if ('str' in obj) {
        const s: string = obj.str;
        // obj: { str: string; num: number }
    } else {
        const _exhaustiveCheck: never = obj;
        // No error. obj: never
    }
};

(obj: { str: string } & { num: number } | { bool: boolean }) => {
    // Union of intersection works as expected.
    if ('str' in obj) {
        const s: string = obj.str;
        // obj: { str: string; num: number }
    } else {
        const proveIsBool: boolean = obj.bool;
        // No error. obj: { bool: boolean }
    }
};

```

<details><summary><b>Output</b></summary>

```ts
"use strict";
(obj) => {
    // Non-Union Intersection type doesn't work as expected.
    if ('str' in obj) {
        const s = obj.str;
        // obj: { str: string } & { num: number }
    }
    else {
        const _exhaustiveCheck = obj;
        // Expected: No error. obj: never
        // Actual: Type '{ str: string; } & { num: number; }' is not assignable to type 'never'.
    }
};
(obj) => {
    // Non-intersection type works as expected.
    if ('str' in obj) {
        const s = obj.str;
        // obj: { str: string; num: number }
    }
    else {
        const _exhaustiveCheck = obj;
        // No error. obj: never
    }
};
(obj) => {
    // Union of intersection works as expected.
    if ('str' in obj) {
        const s = obj.str;
        // obj: { str: string; num: number }
    }
    else {
        const proveIsBool = obj.bool;
        // No error. obj: { bool: boolean }
    }
};

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?ts=3.9.0-dev.20200224&ssl=1&ssc=1&pln=13&pc=1#)

**Comment by RyanCavanaugh** [comment]
Reopening due to #38168
